File: nodejs/worker.js
const amqp = require('amqplib');

const main = async () => {
  const conn = await amqp.connect('amqp://localhost');
  const channel = await conn.createChannel();

  const queue = 'tarefas';

  channel.assertQueue(queue, { durable: true });
  channel.prefetch(1);

  console.log(" [x] Aguardando tarefas ...");
  channel.consume(queue, (msg) => {
    var secs = msg.content.toString().split('.').length - 1;

    console.log(" [x] Recebido %s", msg.content.toString());

    setTimeout(() => {
      console.log(" [x] Pronto!");
      channel.ack(msg);
    }, secs * 1000);
  }, {noAck: false});
};

main();

File: python/worker.py
import time
import pika

RABBITMQ_HOST = 'localhost'
QUEUE = 'tarefas'

params     = pika.ConnectionParameters(RABBITMQ_HOST)
connection = pika.BlockingConnection()
channel    = connection.channel()

channel.queue_declare(queue=QUEUE, durable=True)

def callback(ch, method, properties, body):
    print(" [x] Recebido %r - Trabalhando na tarefa..." % body)
    time.sleep(body.count(b'.'))
    print(" [x] Pronto")
    ch.basic_ack(delivery_tag = method.delivery_tag)

channel.basic_qos(prefetch_count=1)
channel.basic_consume(queue=QUEUE, on_message_callback=callback)

print(' [*] Aguardando mensagens. Para sair, pressione CTRL+C')
channel.start_consuming()